src/protocol_pro_2.cpp

```diff
@@ -235,4 +283,60 @@ void Pro2ProtocolObject::motors_control_loop(int sleeptime) {
   }
 }
 
+void Pro2ProtocolObject::update_params(std::string replacing_key,
+                                       std::string replacing_value) {
+  // open previous config file;
+  std::ifstream robotconfig;
+  std::ofstream newrobotconfig;
+  std::string const HOME = std::getenv("HOME") ? std::getenv("HOME") : ".";
+
+  robotconfig.open(HOME + "/rover.config");
+  newrobotconfig.open(HOME + "/rover.config.tmp");
+  if (robotconfig.is_open()) {
+    std::string line;
+    std::cerr << "Updating config at " + HOME + "/rover.config" << std::endl;
+    while (std::getline(robotconfig, line)) {
+      std::vector<std::string> key =
+          split(line, ":");  // get key-value pair into vector
+      std::vector<std::string> values =
+          split(key[1], " ");         // get all the values from this key
+      if (key[0] == replacing_key) {  // check for key
+        newrobotconfig << replacing_key << ":" << replacing_value
+                       << std::endl;  // output modified value instead
+      } else
+        newrobotconfig << line;  // output original value
+    }
+  }
+  robotconfig.close();
+  newrobotconfig.close();
+  std::string path = HOME + "/rover.config";
+  int n = path.length();
+  char robotconfig_path[n + 1];
+  strcpy(robotconfig_path, path.c_str());
+  path = HOME + "/rover.config.tmp";
+  n = path.length();
+  char newrobotconfig_path[n + 1];
+  strcpy(newrobotconfig_path, path.c_str());
+  if (rename(newrobotconfig_path, robotconfig_path) == 0)
+    puts("Config File Updated");
+  else
+    perror("Error Saving config file");
+}
+std::vector<std::string> Pro2ProtocolObject::split(std::string str,
```

Comment:
It seems to me that this type of text editing operation should be put in a utils file or similar since we may end up using it elsewhere.